bug: ci fails when pushing `chore(release): publish` commit after running `nx release`

When creating a new release using `nx release`, the ci job `publish` fails every time - https://github.com/s1seven/schema-tools/actions/runs/8157244270/job/22296463018

The issue is that the root `package-lock.json` file is not in sync after the updates, causing the ci to fail with the following error:

npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.

Investigate why the updated `package-lock.json` file is not being automatically committed.

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -43,13 +43,13 @@ jobs:
 
     steps:
       - name: Check out repository code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
           ref: ${{ github.head_ref }}
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
 
@@ -64,7 +64,7 @@ jobs:
 
       - name: Derive appropriate SHAs for base and head for `nx affected` commands
         id: set-sha
-        uses: nrwl/nx-set-shas@v3
+        uses: nrwl/nx-set-shas@v4
 
       - name: Install GraphicsMagick
         run: sudo apt-get update && sudo apt-get install -y ghostscript graphicsmagick
@@ -98,13 +98,13 @@ jobs:
 
     steps:
       - name: Check out repository code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
           ref: ${{ github.head_ref }}
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
 
@@ -119,7 +119,7 @@ jobs:
 
       - name: Derive appropriate SHAs for base and head for `nx affected` commands
         id: set-sha
-        uses: nrwl/nx-set-shas@v3
+        uses: nrwl/nx-set-shas@v4
 
       - name: Install dependencies
         run: npm ci
@@ -151,7 +151,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out repository code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Get current version
         id: package-version
@@ -179,7 +179,7 @@ jobs:
     timeout-minutes: 10
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Get branch name (merge)
         if: github.event_name != 'pull_request'

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -160,8 +160,9 @@ jobs:
       - name: Retrieve coverage
         uses: actions/download-artifact@v4
         with:
-          name: coverage-*
+          pattern: coverage-*
           path: ./coverage
+          merge-multiple: true
 
       - name: SonarCloud Scan
         uses: sonarsource/sonarcloud-github-action@master

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -86,7 +86,7 @@ jobs:
       - name: Upload coverage
         uses: actions/upload-artifact@v4
         with:
-          name: coverage
+          name: coverage-${{ matrix.node-version }}
           path: ./coverage
           retention-days: 2
 
@@ -160,7 +160,7 @@ jobs:
       - name: Retrieve coverage
         uses: actions/download-artifact@v4
         with:
-          name: coverage
+          name: coverage-${{ matrix.node-version }}
           path: ./coverage
 
       - name: SonarCloud Scan

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -54,7 +54,7 @@ jobs:
           node-version: ${{ matrix.node-version }}
 
       - name: Cache Node.js modules
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: |
             ~/.npm
@@ -84,7 +84,7 @@ jobs:
           CI: true
 
       - name: Upload coverage
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: coverage
           path: ./coverage
@@ -109,7 +109,7 @@ jobs:
           node-version: ${{ matrix.node-version }}
 
       - name: Cache Node.js modules
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: |
             ~/.npm
@@ -158,7 +158,7 @@ jobs:
         uses: martinbeentjes/npm-get-version-action@main
 
       - name: Retrieve coverage
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
           name: coverage
           path: ./coverage

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -160,7 +160,7 @@ jobs:
       - name: Retrieve coverage
         uses: actions/download-artifact@v4
         with:
-          name: coverage-${{ matrix.node-version }}
+          name: coverage-*
           path: ./coverage
 
       - name: SonarCloud Scan

--- a/.github/workflows/node.yml
+++ b/.github/workflows/node.yml
@@ -39,7 +39,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x]
+        node-version: [18.x, 20.x]
 
     steps:
       - name: Check out repository code

--- a/README.md
+++ b/README.md
@@ -214,11 +214,11 @@ When creating a new release, the following tests need to be updated with the lat
 
 ## Creating new releases
 
-To first see what will be released, run `nx release --dry-run`. 
+To first see what will be released, run `nx release --dry-run`.
 
-Note: releases should only be created on `main`.
+Note: releases should only be created on `main`. Be sure to export a `GITHUB_TOKEN` env variable so the github releases can be done automatically. You should also be logged in to the `npm` cli if creating an `npm` release. To login, use `npm login` from the cli.
 
-To release new versions of the schema-tools packages, run `nx release`. This will automatically bump package version numbers, generate changelogs, create github releases and release on npm.
+To release new versions of the schema-tools packages, run `GITHUB_TOKEN=<token> nx release`. This will automatically bump package version numbers, generate changelogs, create github releases and release on npm.
 
 ## License
 

--- a/nx.json
+++ b/nx.json
@@ -17,7 +17,10 @@
       }
     },
     "version": {
-      "conventionalCommits": false
+      "conventionalCommits": false,
+      "generatorOptions": {
+        "skipLockFileUpdate": true
+      }
     },
     "git": {
       "commit": true,

